### PR TITLE
Add Win64 build script and full OpenCL pipeline

### PR DESCRIPTION
**Pull request by nyanmisaka**
**FFmpeg**
- New upstream version 4.4.1

**Build script**
- Only disable x86 ASM in Dav1D if nasm is outdated.
- Add build script and github workflow for Windows win64.
- Update Intel VAAPI deps and Dav1D.
- Add support for Ubuntu Impish 21.10.
- Drop support for **EOL Ubuntu versions([18.10, 19.04, 19.10, 20.10]**(https://wiki.ubuntu.com/Releases)).

**HWA** pipeline & codecs
- Refactor AMD AMF codes and add HEVC 10bit encoding support.(Win64)
- Add OpenCL based video scaler/resizer and pixel format converter.
- Refactor OpenCL based tonemap filter, add LUT for faster de/linearize and more SW format.
- Refactor OpenCL based overlay filter, add PGS graphical subtitle support.
- Add D3D11-OpenCL interop for AMD.(Win64)
- Add QSV(D3D11)-OpenCL interop for Intel.(Win64)
- Add D3D11 suppport for QSV.(Win64)
- Add QSV yuv444 8/10bit decoding.
- Add miscellaneous fixes for QSV and Nvidia from upstream.
- Add options to set input metadata in VAAPI VPP tonemap.
(instead of throwing error if the embedded one does not exist)
metadata syntax: `G(%hu|%hu)B(%hu|%hu)R(%hu|%hu)WP(%hu|%hu)L(%u|%u)` and `CLL(%hu)FALL(%hu)`

**SW**
- Fix for webvttenc when using segement muxer.
- Fix for HEVC 10bit HDR decoding in bit stream filtering.
- Add alphasrc source video filter for textual sub2video.
- Add sub2video option to subtitles/ass filters.

### CODE REVIEW

**Comment by shaino** [comment]
Great job. @nyanmisaka 
I 'docker pull' the latest nightly version of linuxserver/jellyfin (10.8.0) and found that PGS subtitles still could not be displayed using OpenCL tone mapping.
ffmpeg4.4-1 build 10102021
intel-media-va-driver-non-free 21.2.2 or 21.3.1
intel-opencl-icd 21.29.20389 or 21.33.20678
thx

**Comment by nyanmisaka** [comment]
> Great job. @nyanmisaka I 'docker pull' the latest nightly version of linuxserver/jellyfin (10.8.0) and found that PGS subtitles still could not be displayed using OpenCL tone mapping. ffmpeg4.4-1 build 10102021 intel-media-va-driver-non-free 21.2.2 or 21.3.1 intel-opencl-icd 21.29.20389 or 21.33.20678 thx

Extra changes in jellyfin are required.

**Review by joshuaboniface — APPROVED** [review]

File: Dockerfile.win64.make [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/make
+DISTRO=ubuntu:hirsute
+FF_REV=1
+.PHONY: Dockerfile
+Dockerfile: Dockerfile.win64.in
+	sed 's/DISTRO/$(DISTRO)/; s/FFMPEG_REV/$(FF_REV)/' $< > $@ || rm -f $@
+clean:
+	rm -f Dockerfile
```
Comment (h1dden-da3m0n) [inline Dockerfile.win64.make:8]: any particular reason why you went with a make script to sed this over using `ARG` parameters in the Dockerfile?

Comment (nyanmisaka) [reply Dockerfile.win64.make:8]: It was just copied from `Dockerfile.make`.
https://github.com/jellyfin/jellyfin-ffmpeg/blob/jellyfin/Dockerfile.make

Comment (h1dden-da3m0n) [reply Dockerfile.win64.make:8]: I see, I have no strong opinion on it I just question why would you do that over using ARGs in the first place :wink:
Thank you for the response!
(considering this resolved)

**Comment by nyanmisaka** [comment]
This PR can be merged now.

~~Still WIP. Dont merge.~~

**Comment by nyanmisaka** [comment]
This PR can be merged now.

**Review by joshuaboniface — APPROVED** [review]

**Comment by nyanmisaka** [comment]
I think it's ready for the first beta.

**Review by joshuaboniface — APPROVED** [review]
YOLO

**Review by crobibero — APPROVED** [review]
#yoloyolo

**Pull request merged**